src/common/AbstractObject.cpp

```diff
@@ -666,44 +667,80 @@ void AbstractObject::addAlias(const std::string & authority, const std::string &
 	if (partialObject != nullptr)
 		throw invalid_argument("The wrapped gsoap proxy must not be null");
 
-	eml20__ObjectAlias* alias = soap_new_eml20__ObjectAlias(gsoapProxy2_0_1->soap, 1);
-	alias->authority = soap_new_std__string(gsoapProxy2_0_1->soap, 1);
-	alias->authority->assign(authority);
-	alias->Identifier = title;
-	static_cast<resqml2__AbstractResqmlDataObject*>(gsoapProxy2_0_1)->Aliases.push_back(alias);
+	if (gsoapProxy2_0_1 != nullptr) {
+		eml20__ObjectAlias* alias = soap_new_eml20__ObjectAlias(gsoapProxy2_0_1->soap, 1);
+		alias->authority = gsoap_resqml2_0_1::soap_new_std__string(gsoapProxy2_0_1->soap, 1);
+		alias->authority->assign(authority);
+		alias->Identifier = title;
+		gsoapProxy2_0_1->Aliases.push_back(alias);
+	}
+	else {
+		eml21__ObjectAlias* alias = soap_new_eml21__ObjectAlias(gsoapProxy2_1->soap, 1);
+		alias->authority = authority;
+		alias->Identifier = title;
+		gsoapProxy2_1->Aliases.push_back(alias);
+	}
 }
 
 unsigned int AbstractObject::getAliasCount() const
 {
 	if (partialObject != nullptr)
 		throw invalid_argument("The wrapped gsoap proxy must not be null");
 
-	return static_cast<resqml2__AbstractResqmlDataObject*>(gsoapProxy2_0_1)->Aliases.size();
+	size_t count = 0;
+	
+	if (gsoapProxy2_0_1 != nullptr) {
```

Comment:
Ternary operator or "else if". "else if" would have been better regarding my previous comment.
In case of else if, a final else should throw an exception .
not a blocking comment at all.

Reply:
Ok.